src/core/thread/mle.cpp

```diff
@@ -1484,6 +1485,12 @@ void Mle::HandleStateChanged(otChangedFlags aFlags)
 #endif
     }
 
+    if (mChildUpdateRequestState == kChildUpdateRequestPending)
+    {
+        mChildUpdateAttempts = 0;
+        mChildUpdateRequestTimer.Start(kChildUpdateRequestPendingDelay);
```

Comment:
How about moving the above two lines to after where we set the `mChildUpdateRequestState = kChildUpdateRequestPending;` in two cases above in  this method?

This way, we ensure that the timer is only restarted when the state does change. The reason is if we receive a `HandeStateChanged` with another flag (that does not require a "Child Update") while we are in Pending state, I don't think we need to restart the timer.

Thoughts?

Reply:
oh. yes. it make senses.